core/src/main/java/org/axonframework/eventhandling/saga/AbstractSagaManager.java

```diff
@@ -147,30 +144,15 @@ protected boolean matchesSegment(Segment segment, String sagaId) {
      */
     protected abstract Set<AssociationValue> extractAssociationValues(EventMessage<?> event);
 
-    private boolean doInvokeSaga(EventMessage event, Saga<T> saga) {
+    private boolean doInvokeSaga(EventMessage event, Saga<T> saga) throws Exception {
         try {
             return saga.handle(event);
         } catch (Exception e) {
-            if (suppressExceptions) {
-                logger.error(format("An exception occurred while a Saga [%s] was handling an Event [%s]:",
-                                    saga.getClass().getSimpleName(), event.getPayloadType().getSimpleName()), e);
-                return true;
-            } else {
-                throw e;
-            }
+            sagaInvocationErrorHandler.onError(e, event, saga);
+            return true;
         }
     }
 
-    /**
```

Comment:
Isn't dropping this backwards compatibility an issue, as it would break users which move to the new version of Axon?

Isn't it possible to keep this `suppressExceptions` check and switch between the provided error handler if it's false, and to a default 'no op error handler' if it's true?
That way we'll not break backwards compatibility.

Reply:
That sounds solid, thumbs up for that approach.